### PR TITLE
Track envvars

### PR DESCRIPTION
**Pull request by hadley**

### CODE REVIEW

File: R/tests.R [hunk 1]
```diff
@@ -10,23 +10,18 @@ renv_tests_scope <- function(packages = character(), project = NULL, envir = par
...
   renv_tests_init()
   renv_tests_scope_repos(envir = envir)
+  renv_scope_envvars(R_LIBS_USER = NULL, envir = envir)
 
   # ensure that attempts to restart are a no-op
   if (renv_rstudio_available())
     options(restart = function(...) TRUE)
 
-  # save local repositories
-  Sys.setenv(RENV_PATHS_LOCAL = renv_tests_path("local"))
```
Comment (hadley) [inline R/tests.R:19]: This was only used by two tests, so they now do it themselves.

File: R/tests.R [hunk 2]
```diff
@@ -10,23 +10,18 @@ renv_tests_scope <- function(packages = character(), project = NULL, envir = par
 
   renv_tests_init()
   renv_tests_scope_repos(envir = envir)
+  renv_scope_envvars(R_LIBS_USER = NULL, envir = envir)
```
Comment (hadley) [inline R/tests.R:13]: `init()` changes this so this ensures we reset at the end of the test.

File: R/tests.R [hunk 3]
```diff
@@ -53,20 +48,13 @@ renv_tests_scope <- function(packages = character(), project = NULL, envir = par
...
-  Sys.unsetenv("RENV_PROFILE")
-  Sys.unsetenv("RENV_PROJECT")
   Sys.unsetenv("RENV_PATHS_ROOT")
   Sys.unsetenv("RENV_PATHS_LIBRARY")
   Sys.unsetenv("RENV_PATHS_LIBRARY_ROOT")
   Sys.unsetenv("RENV_PATHS_LOCAL")
   Sys.unsetenv("RENV_PATHS_LOCKFILE")
   Sys.unsetenv("RENV_PATHS_RENV")
 
-  Sys.unsetenv("RENV_PYTHON")
```
Comment (hadley) [inline R/tests.R:65]: Now handled by `renv_test_scope_python()`

File: R/tests.R [hunk 4]
```diff
@@ -53,20 +48,13 @@ renv_tests_scope <- function(packages = character(), project = NULL, envir = par
 
 renv_tests_init_envvars <- function() {
 
-  Sys.unsetenv("RENV_PROFILE")
```
Comment (hadley) [inline R/tests.R:56]: Now set/restored in the one profile test that needs it.

File: tests/testthat/helper-aaa.R [hunk 5]
```diff
@@ -60,11 +60,17 @@ renv_test_state <- function() {
   opts$repos[opts$repos == "@CRAN@"] <- "https://cloud.r-project.org"
   opts <- opts[csort(names(opts))]
 
+  envvars <- as.list(Sys.getenv())
+  envvars <- envvars[grep("^RENV_DEFAULT_", names(envvars), invert = TRUE)]
```
Comment (hadley) [inline tests/testthat/helper-aaa.R:64]: Let me know if you think we should be tracking these env vars, but they didn't seem that important, and frequently changed.

File: R/tests.R (hunk 2 above)
Comment (kevinushey) [reply R/tests.R:13]: Maybe worth adding as a comment in the code so we remember the reason?

File: tests/testthat/helper-aaa.R (hunk 5 above)
Comment (kevinushey) [reply tests/testthat/helper-aaa.R:64]: This looks fine to me too.

**Review by kevinushey — APPROVED** [review]

**Pull request merged**